test/shared-class.test.js

```diff
@@ -243,6 +268,26 @@ describe('SharedClass', function() {
       expect(methods).to.not.contain(DYN_METHOD_NAME);
     });
   });
+
+  describe('sharedClass.disableMethodByName(methodName)', function() {
+    it('excludes disabled static methods from the method list', function() {
+      var METHOD_NAME = 'testMethod';
+      var sc = new SharedClass('SomeClass', SomeClass);
+      var sm = sc.defineMethod(METHOD_NAME, {isStatic: true});
+      sc.disableMethodByName(METHOD_NAME);
+      var methods = sc.methods().map(getName);
+      expect(methods).to.not.contain(METHOD_NAME);
+    });
+
+    it('excludes disabled prototype methods from the method list', function() {
+      var INST_METHOD_NAME = 'prototype.instTestMethod';
+      var sc = new SharedClass('SomeClass', SomeClass);
+      var sm = sc.defineMethod('instTestMethod', {isStatic: false});
+      sc.disableMethodByName(INST_METHOD_NAME);
+      var methods = sc.methods().map(getName);
+      expect(methods).to.not.contain(INST_METHOD_NAME);
```

Comment:
Are we sure `getName` returns the name with the `prototype` suffix? I think the opposite is the case.

To be sure, could you please add another check

```js
expect(methods).to.not.contain('instTestMethod');
```